### PR TITLE
[WFCORE-5285] Utilize o.w.common.Assert for Null-Checks (security)

### PR DESCRIPTION
**Pull request by boris-unckel**
Fixing https://issues.redhat.com/browse/WFCORE-5285

Original PR was too big to verify. It has been splitted up, here: security

### CODE REVIEW

File: core-security/implementation/pom.xml [hunk 1]
```diff
@@ -59,6 +59,11 @@
             <optional>true</optional>
         </dependency>
 
+        <dependency>
+            <groupId>org.wildfly.common</groupId>
```
Comment (jmesnil) [inline core-security/implementation/pom.xml:63]: @darranl do you see any issue with adding this dependency?

Comment (darranl) [reply core-security/implementation/pom.xml:63]: I think the dependency is find but it looks like the module does not have the dependency so it is odd this is not resulting in an error.

Comment (darranl) [reply core-security/implementation/pom.xml:63]: Actually there are a number of test failures, I would expect this module to also need the dependency adding:

https://github.com/wildfly/wildfly-core/blob/master/core-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/as/core-security/main/module.xml

**Review by jmesnil — CHANGES_REQUESTED** [review]
could you please add the dependency to the `org.jboss.as.core-security` module?

**Comment by boris-unckel** [comment]
@jmesnil all tests are successful, except "Full integration - Windows" where the CI failed. Please check for approval.

**Review by jmesnil — APPROVED** [review]

**Pull request merged**